### PR TITLE
Fix Builder.withTraceNothing() ignoring @Traced

### PR DESCRIPTION
**Pull request by sjoerdtalsma**
See #107 for details.

### CODE REVIEW

**Comment by pavolloffay** [comment]
it seems that test failed

**Comment by sjoerdtalsma** [comment]
> it seems that test failed

I noticed: I just changed the one line from the web api, will have to checkout the project now I gues :wink:

File: opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractJettyTest.java [hunk 1]
```diff
@@ -107,14 +106,7 @@ public void after() {
...
     public static void resetGlobalTracer() {
-        try {
-            Field globalTracerField = GlobalTracer.class.getDeclaredField("tracer");
-            globalTracerField.setAccessible(true);
-            globalTracerField.set(null, NoopTracerFactory.create());
-            globalTracerField.setAccessible(false);
-        } catch (Exception e) {
-            throw new RuntimeException("Error resetting " + GlobalTracer.class, e);
-        }
+        GlobalTracerTestUtil.resetGlobalTracer();
```
Comment (sjoerdtalsma) [inline opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractJettyTest.java]: :wink:

File: opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractServerWithTraceNothingTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        response.close();
+        await().until(finishedSpansSizeEquals(1));
+
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        Assert.assertEquals(1, mockSpans.size());
+        assertOnErrors(mockTracer.finishedSpans());
+    }
+
+    @Test
+    public void testNotExistingURL() {
```
Comment (pavolloffay) [inline opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractServerWithTraceNothingTest.java]: please remove this test it's not necessary here

File: opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractServerWithTraceNothingTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        Response response = client.target(url("/hello/1?q=a"))
+                .request()
+                .get();
+        response.close();
+        List<MockSpan> mockSpans = mockTracer.finishedSpans();
+        Assert.assertEquals(0, mockSpans.size());
+    }
+
+    @Test
+    public void testTracedFalseMethod() {
```
Comment (pavolloffay) [inline opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractServerWithTraceNothingTest.java]: can be removed it's  the same as the previous test `testUnAnnotatedMethod`

Comment (sjoerdtalsma) [reply opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractServerWithTraceNothingTest.java]: It isn't is it? In the one situation @Traced is null, here it is non-null with value=false ?

Comment (pavolloffay) [reply opentracing-jaxrs2-itest/common/src/main/java/io/opentracing/contrib/jaxrs2/itest/common/AbstractServerWithTraceNothingTest.java]: Isn't it calling the same URL?

**Comment by pavolloffay** [comment]
I have pushed 0.2.1

**Pull request merged**